src/content/en/2021/javascript.md

```diff
@@ -1,20 +1,1544 @@
+  sheets_gid="821396474",
+  sql_file="compression_method_by_3p.sql"
+  )
+}}
+
+59% of third-party scripts are gzipped and 30% are brotli compressed. Looking at first-party scripts, these are 52% with gzip compression but only 32% with brotli. There are still 11% of third-party scripts that do not have any compression method defined for the resources.
+
+{{ figure_markup(
+  image="uncompressed-first-third-party.png",
+  caption="Uncompressed resources for first party vs third party.",
+  description="",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=1043138204&format=interactive",
+  sheets_gid="1942216454",
+  sql_file="compression_none_by_bytes.sql"
+  )
+}}
+
+
+The way preloading all our requests negates the impact of the attribute, a similar problem occurs when we try to optimise and compress some resource that doesn’t need compression and is already small. It is observed that 90% of uncompressed third party JS resources are &lt; 5 KB, and uncompressed first-party resources less than 10kb are only 8%. 
+
+
+### Minification
+
+When working towards reducing the script parsing time, we tend to focus on all the opportunities to make our code smaller and relevant. One such idea is to minify the files and bring down the bytes usage.
+
+The lighthouse report also [highlights the unminified JS being used](https://web.dev/unminified-javascript/) and lists these unminified files and the opportunities to save.
+
+{{ figure_markup(
+  image="unminified-js-audit-scores.png",
+  caption="Percentage distribution of unminified JS audit scores.",
+  description="",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=1572896641&format=interactive",
+  sheets_gid="1539653841",
+  sql_file="lighthouse_unminified_js.sql"
+  )
+}}
+
+It is noted that 67% of mobile pages have an "unminified JS" score between 0.9 and 1.0, meaning that they have a few unminified JS resources.
+
+ This is an improvement of 10% as compared to the results from the last year (2020).
+
+To dive deeper into understanding how many bytes per page are unminified.
+
+{{ figure_markup(
+  image="unminified-js-bytes.png",
+  caption="Percentage distribution of unminified JS bytes per page.",
+  description="",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=1069289665&format=interactive",
+  sheets_gid="1511556407",
+  sql_file="lighthouse_unminified_js_bytes.sql"
+  )
+}}
+
+And it is found that 57% of mobile pages have 0 KB of unminified JS whereas 18% of those have 0-10 kilobytes of unminified JS.
+
+The first party vs. third party analysis in this case shows that 82% of the average mobile page's unminified JS bytes actually come from first-party scripts.
```

Comment:
That's not what the image below shows. It shows 82% of unminified bytes come from third parties.

Reply:
Thanks for catching this. Fixed!